source/operations/concepts/healing.rst

```diff
@@ -0,0 +1,110 @@
+.. _minio-concepts-healing:
+
+==============
+Object Healing
+==============
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 1
+
+What is healing?
+----------------
+
+Healing is MinIO's ability to restore an object that has been damaged, corrupted, or partially lost.
+The loss can come from random corruptions in the form of 
+- :ref:`bit rot <minio-ec-bitrot>`
+- the loss of one or more drives due to failure or outages
+- the corruption or loss of one or more shards of an object for other reasons
+
+Healing and Erasure Coding
+--------------------------
+
+The ability of MinIO to restore a damaged object relates directly to the following:
+
+- total number of drives in the :ref:`erasure set <minio-erasure-coding>` where the object exists
+- :ref:`parity setting <minio-ec-parity>` for the erasure set
+
+  Parity refers to the number of extra shards MinIO create when writing the object.
+  For example, an erasure set may have eight total drives and use three drives during a write for parity.
+  In that scenario, MinIO splits the object into five parts written to five of the drives.
+  MinIO uses the remaining three drives in the erasure set for parity shards.
+
+  When MinIO needs to provide the object, it looks for the data shards for the object.
+  If any of the data shards are missing or damaged, MinIO uses the parity shards to restore the object.
+  For this scenario, up to three of data shard parts can be lost or damaged and MinIO can still successfully restore and serve the object. 
+- number of drives available with intact parts of the object
+
+  The number of drives available with intact data or parity shards of the object must meet or exceed the number of drives used for data shards in the erasure set.
+  In the scenario above, five drives with intact shards must be online and available for MinIO to successfully serve the object.
+
+When does MinIO heal an object?
+-------------------------------
+
+MinIO has a robust system for healing objects.
+
+Healing during ``GET`` requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO automatically checks the consistency of an object's data shards each time you request an object with a ``GET`` operation.
+
+If all data shards are found intact, MinIO takes no further action on the object and serves the object from the data shards.
+
+If MinIO finds missing or damaged data shards, MinIO utilizes the available parity shards to heal the object before serving it as part of the ``GET`` operation.
+There **must** be a parity shard available for each lost or damaged data shard.
+
+Healing with the object scanner
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO utilizes an :ref:`object scanner <minio-concepts-scanner>` to perform a number of tasks related to objects.
+One of these tasks checks the integrity of objects and, if found damaged or corrupted, heals them.
+
+On each scanning pass, MinIO selects one out of every 1,024 objects to check.
```

Comment:
This might imply we check 1 object per pass/cycle. Not sure how best to address that.

"MinIO checks a portion of the total namespace on each pass, using a filter ratio to determine which objects to check or skip."

Reply:
Tweaked your suggestion, but running with it.